File: my_flutter_gen/lib/feature/auth/domain/usecase/login_usecase.dart
import 'package:my_flutter_gen/core/network/api_result.dart';
import 'package:my_flutter_gen/feature/auth/domain/model/user.dart';
import 'package:my_flutter_gen/feature/auth/domain/repository/auth_repository.dart';

/// 登录用例 —— 封装登录的业务逻辑
class LoginUseCase {
  final AuthRepository _repository;

  const LoginUseCase(this._repository);

  /// 执行登录
  Future<ApiResult<User>> call({
    required String email,
    required String password,
  }) {
    return _repository.login(email: email, password: password);
  }
}

File: MyComposeProject/app/src/main/java/com/louis/mycomposeproject/domain/usecase/LoginUseCase.kt
package com.louis.mycomposeproject.domain.usecase

import com.louis.mycomposeproject.domain.model.User
import com.louis.mycomposeproject.domain.repository.AuthRepository

class LoginUseCase(private val repository: AuthRepository) {
    suspend operator fun invoke(email: String, password: String): Result<User> {
        if (email.isBlank() || password.isBlank()) {
            return Result.failure(Exception("Email and password cannot be empty"))
        }
        return repository.login(email, password)
    }
}
